### DISCUSSION TITLE
Sobre DataSetAdapter - Ele foi removido ?

### DISCUSSION THREAD

**Discussion by AlexhaifaGarcia** [viniciussanchez/RESTRequest4Delphi#204] [Q&A] [answered]
Olá Pessoal, tudo bem?

Neste código abaixo:

resp := TRequest.New.BaseURL('http://localhost:9000')
            .Resource('messages')
            .BasicAuthentication('testserver', 'testserver')
            .AddParam('cod_usuario', cod_usuario_logado.ToString)
            .AddParam('cod_mensagem', '0')
            .Accept('application/json')
            .DataSetAdapter(TabMensagem)
            .Get;

O delphi da o seguinte erro: Undeclared identifier: 'DataSetAdapter'

No video aula que estou fazendo, ele tem isso como exemplo, poderiam me orientar a respeito
Grato

**Comment by viniciussanchez** [comment, marked as answer]
Boa tarde...
Foi trocado por adapters:
![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/16382981/1cff4322-b15d-4706-91d8-14290f87c2b7)
![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/16382981/91d87c58-c5fa-41a6-b135-62787d393cd4)
Adicionando novas possibilidades de expansão ao projeto...

**Reply by AlexhaifaGarcia** [reply]
Muito obrigado pelo retorno.
Vou atras do ajuste agora.

**Reply by moises-sistemas** [reply]
para quem não estiver conseguindo utilizar o TdatasetSerializeAdapter segue o link da solução que funcionou para mim. https://github.com/viniciussanchez/dataset-serialize/discussions/213

**Comment by cyber233** [comment]
Bom dia, mesmo add as Units, dando erro.
![error](https://github.com/user-attachments/assets/855f88dd-e7f8-4fd4-9940-1f1d888c9631)

**Comment by cyber233** [comment]
comentei duas linhas e está funcionando de boa...
![erro2](https://github.com/user-attachments/assets/9ea9ebfb-d60f-46ad-85b4-6b86eb87dba7)

**Reply by viniciussanchez** [reply]
Você está usando a última release liberada? Não use o código na raiz que está em manutenção, sempre use a release
